sdk_container/src/third_party/coreos-overlay/sys-apps/ignition/files/0019-Detect-multipart-mime-userdata.patch

```diff
@@ -0,0 +1,291 @@
++++ b/vendor/github.com/flatcar/ignition/config/v2_2/config.go
+@@ -35,6 +35,8 @@ func Parse(rawConfig []byte) (types.Config, report.Report, error) {
+ 		return types.Config{}, report.Report{}, errors.ErrCloudConfig
+ 	} else if isScript(rawConfig) {
+ 		return types.Config{}, report.Report{}, errors.ErrScript
++	} else if isMultipartMime(rawConfig) {
++		return types.Config{}, report.Report{}, errors.ErrMultiPart
+ 	}
+ 
+ 	var err error
+diff --git a/vendor/github.com/flatcar/ignition/config/v2_3/cloudinit.go b/vendor/github.com/flatcar/ignition/config/v2_3/cloudinit.go
+index 3b83fbd9..5d42c3f3 100644
+--- a/vendor/github.com/flatcar/ignition/config/v2_3/cloudinit.go
++++ b/vendor/github.com/flatcar/ignition/config/v2_3/cloudinit.go
+@@ -24,6 +24,15 @@ import (
+ 	"unicode"
+ )
+ 
++func isMultipartMime(userdata []byte) bool {
++	header := strings.SplitN(string(decompressIfGzipped(userdata)), "\n", 2)[0]
++
++	// Trim trailing whitespaces
++	header = strings.TrimRightFunc(header, unicode.IsSpace)
++
++	return strings.Contains(header, "Content-Type: multipart/mixed;")
++}
++
+ func isCloudConfig(userdata []byte) bool {
+ 	header := strings.SplitN(string(decompressIfGzipped(userdata)), "\n", 2)[0]
+ 
+diff --git a/vendor/github.com/flatcar/ignition/config/v2_3/config.go b/vendor/github.com/flatcar/ignition/config/v2_3/config.go
+index 4693547a..2172f821 100644
+--- a/vendor/github.com/flatcar/ignition/config/v2_3/config.go
++++ b/vendor/github.com/flatcar/ignition/config/v2_3/config.go
+@@ -34,6 +34,8 @@ func Parse(rawConfig []byte) (types.Config, report.Report, error) {
+ 		return types.Config{}, report.Report{}, errors.ErrCloudConfig
+ 	} else if isScript(rawConfig) {
+ 		return types.Config{}, report.Report{}, errors.ErrScript
++	} else if isMultipartMime(rawConfig) {
++		return types.Config{}, report.Report{}, errors.ErrMultiPart
+ 	}
+ 
+ 	var err error
+diff --git a/vendor/github.com/flatcar/ignition/config/v2_4/cloudinit.go b/vendor/github.com/flatcar/ignition/config/v2_4/cloudinit.go
+index 0c428e62..b7b91f1f 100644
+--- a/vendor/github.com/flatcar/ignition/config/v2_4/cloudinit.go
++++ b/vendor/github.com/flatcar/ignition/config/v2_4/cloudinit.go
+@@ -24,6 +24,15 @@ import (
+ 	"unicode"
+ )
+ 
++func isMultipartMime(userdata []byte) bool {
++	header := strings.SplitN(string(decompressIfGzipped(userdata)), "\n", 2)[0]
++
++	// Trim trailing whitespaces
++	header = strings.TrimRightFunc(header, unicode.IsSpace)
++
++	return strings.Contains(header, "Content-Type: multipart/mixed;")
++}
++
+ func isCloudConfig(userdata []byte) bool {
+ 	header := strings.SplitN(string(decompressIfGzipped(userdata)), "\n", 2)[0]
+ 
+diff --git a/vendor/github.com/flatcar/ignition/config/v2_4/config.go b/vendor/github.com/flatcar/ignition/config/v2_4/config.go
+index e6120bda..b76deeec 100644
+--- a/vendor/github.com/flatcar/ignition/config/v2_4/config.go
++++ b/vendor/github.com/flatcar/ignition/config/v2_4/config.go
```

Comment:
Do we really need to patch the vendor folder? Maybe we could move these checks to our `0006-config-v3_5-convert-ignition-2.x-to-3.x.patch` and to return early when a cloud config or script is found.

Reply:
no. I was on auto pilot when I made these changes. Searched for places where we added cloud-config and script handling and added multipart mime. I'll refrain from making changes at night :smile:

Reply:
I added the changes to patch 13.